### PR TITLE
Change for TEMPO converter to take real NO2 data

### PR DESCRIPTION
**Pull request by jeromebarre**
## Description

Change for TEMPO converter to take real NO2 data.
The prints are necessary as this ensures quick debugging. The product is still in the early adopter stage and has some inconsistencies in masks.

## Issue(s) addressed
https://github.com/JCSDA-internal/ioda-converters/issues/1415

## Dependencies

List the other PRs that this PR is dependent on:
None

## Impact

None

## Checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] I have run the unit tests before creating the PR

### CODE REVIEW

**Comment by PatNichols** [comment]
@jeromebarre is this in it's final form for review?

**Comment by jeromebarre** [comment]
@PatNichols For this quarter yes

**Review by PatNichols — APPROVED** [review]
@jeromebarre Thanks

File: src/compo/tempo_nc2ioda.py [hunk 1]
```diff
@@ -72,8 +72,10 @@ def _read(self):
         for f in self.filenames:
             ncd = nc.Dataset(f, 'r')
 
-            # conversion factor fron constants
-            conv = cm2m2 * molarmass[self.varname] / Na
+            print('FILENAME: ', f)
```
Comment (mer-a-o) [inline src/compo/tempo_nc2ioda.py:75]: is this intentional?

File: src/compo/tempo_nc2ioda.py [hunk 2]
```diff
@@ -187,60 +197,93 @@ def _read(self):
                 exit()
 
             # clean data
-            neg_obs = obs > 0.0
-            nan_obs = obs != np.nan
+            neg_obs = ((obs > 0.0) & (err > 0.0))
+            nan_obs = ((obs != np.nan) & (err != np.nan))
```
Comment (mer-a-o) [inline src/compo/tempo_nc2ioda.py:201]: Do you need `isnan` here?

File: src/compo/tempo_nc2ioda.py [hunk 3]
```diff
@@ -187,60 +197,93 @@ def _read(self):
...
             cln = np.logical_and(neg_obs, nan_obs)
 
             # final flag before sending this to ioda engines
             flg = np.logical_and(flg, cln)
 
+            # print before compression
```
Comment (mer-a-o) [inline src/compo/tempo_nc2ioda.py:207]: Are these print statements here for debugging?

File: src/compo/tempo_nc2ioda.py [hunk 4]
```diff
@@ -187,60 +197,93 @@ def _read(self):
...
-                self.outdata[self.varDict[iodavar]['qcKey']] = np.concatenate(
-                    (self.outdata[self.varDict[iodavar]['qcKey']], qc_flag[flg]))
-
-            first = False
+            if np.shape(lats)[0] > 0:
+                preslev = np.flip(preslev, axis=1)
+                avg_kernel = np.flip(avg_kernel, axis=1)
+
+                # print after compression
+                print('AFTER COMPRESSION')
```
Comment (mer-a-o) [inline src/compo/tempo_nc2ioda.py:240]: Same here

File: src/compo/tempo_nc2ioda.py (hunk 1 above)
Comment (jeromebarre) [reply src/compo/tempo_nc2ioda.py:75]: yes

File: src/compo/tempo_nc2ioda.py (hunk 2 above)
Comment (jeromebarre) [reply src/compo/tempo_nc2ioda.py:201]: Not sure if isnan would work on creating mask like here. If you think it does please provide the tested coding suggestion/example.

File: src/compo/tempo_nc2ioda.py (hunk 3 above)
Comment (jeromebarre) [reply src/compo/tempo_nc2ioda.py:207]: I'd like to keep them here as default as the TEMPO current format is full of inconsistencies in masking. See PR description.

File: src/compo/tempo_nc2ioda.py (hunk 4 above)
Comment (jeromebarre) [reply src/compo/tempo_nc2ioda.py:240]: see above and PR description.

**Review by mer-a-o — APPROVED** [review]

**Pull request merged**